app/views/moderator/post/posts/confirm_delete.html.erb

```diff
@@ -7,31 +7,56 @@
 <%= form_tag(delete_moderator_post_post_path, :style => "clear: both;", :class => "simple_form") do %>
   <div class="input">
     <label for="reason">Reason</label>
-    <%= text_field_tag "reason", @reason, size: 50 %>
+    <%= text_field_tag "reason", @reason %>
   </div>
 
   <% if @post.parent_id %>
-    <div class="input">
-      <label for="move_favorites">
-        <%= check_box_tag "move_favorites", true, true %>
-        Move favorites to parent?
-      </label>
-      <label for="copy_tags">
-        <%= check_box_tag "copy_tags", false, false %>
-        Merge tags into parent?
-      </label>
-      <label for="copy_sources">
-        <%= check_box_tag "copy_sources", false, false %>
-        Merge sources into parent?
-      </label>
+    <div class="post_delete_options">
       <%= PostPresenter.preview(@post.parent, tags: 'status:any', no_blacklist: true) %>
+      <div class="input">
+        <label for="move_favorites">
+          <%= check_box_tag "move_favorites", true, true %>
+          Move favorites to parent?
+        </label>
+        <label for="copy_tags">
+          <%= check_box_tag "copy_tags", false, false %>
+          Merge tags into parent?
+        </label>
+        <label for="copy_sources">
+          <%= check_box_tag "copy_sources", false, false %>
+          Merge sources into parent?
+        </label>
+      </div>
     </div>
   <% end %>
 
   <%= submit_tag "Delete" %>
   <%= submit_tag "Cancel" %>
+  <%= button_tag "Clear", :type => "button", :id => "delreason-clear" %>
 <% end %>
 
+<div id="delreason-prebuilt" class="simple_form">
+<b>Reason</b>
+<% Danbooru.config.deletion_reasons.each do |deletion_reason| %>
+  <% if deletion_reason.nil? || deletion_reason.empty? %>
+    <br />
+  <% else %>
+    <a class="button delreason-button" data-text="<%= deletion_reason.gsub(/%(PARENT|OTHER)_ID%/, "%ID%") %>">
+      <%= deletion_reason
+            .gsub("%PARENT_ID%", "<input type=\"text\" value=\"" + (@post.parent_id || "").to_s + "\" />")
+            .gsub("%OTHER_ID%", "<input type=\"text\" />")
+            .html_safe
+      %>
+    </a>
+  <% end %>
+<% end %>
+</div>
+
+<div id="deletion-reason-suggestions"></div>
+<%= javascript_tag nonce: true do -%>
```

Comment:
In order to keep on page javascript to a minimum, the usual method of doing this is to check the page ids on page ready and determine if the code should run. The reason this is done is because the main pack might load after the the page has finished parsing, and if that happens then this code runs before the main pack exists and has executed.

Trying to prepare for deferred loading of JS as much as possible.

The other routines have example code for how this is done. It's a bit messy and has some problems with execution order for shared pieces, but it keeps JS all in one place.